autoPyTorch/evaluation/abstract_evaluator.py

```diff
@@ -0,0 +1,533 @@
+        return False
+
+    def get_additional_run_info(self) -> None:  # pylint: disable=R0201
+        return None
+
+
+def fit_and_suppress_warnings(logger: PicklableClientLogger, model: BaseEstimator,
+                              X: Dict[str, Any], y: Any
+                              ) -> BaseEstimator:
+    def send_warnings_to_log(message, category, filename, lineno,
+                             file=None, line=None) -> None:
+        logger.debug('%s:%s: %s:%s',
+                     filename, lineno, category.__name__, message)
+        return
+
+    with warnings.catch_warnings():
+        warnings.showwarning = send_warnings_to_log
+        model.fit(X, y)
+
+    return model
+
+
+# TODO: adjust for differences in image and tabular data, also maybe time series but its low priority
+class AbstractEvaluator(object):
+    def __init__(self, backend: Backend, queue: Queue, metric: List[autoPyTorchMetric],
+                 configuration: Optional[Configuration] = None,
+                 seed: int = 1,
+                 output_y_hat_optimization: bool = True,
+                 num_run: Optional[int] = None,
+                 include: Optional[Dict[str, Any]] = None,
+                 exclude: Optional[Dict[str, Any]] = None,
+                 disable_file_output: bool = False,
+                 init_params: Optional[Dict[str, Any]] = None,
+                 budget: Optional[float] = None,
+                 budget_type: Optional[str] = None,
+                 logger_port: Optional[int] = None) -> None:
+
+        self.starttime = time.time()
+
+        self.configuration = configuration
+        # TODO: adjust backend for autopytorch backend
+        self.backend = backend
+        self.queue = queue
+
+        # TODO: use autopytorch datasets once PR is passed
+        self.datamanager: BaseDataset = self.backend.load_datamanager()
+        self.include = include
+        self.exclude = exclude
+
+        self.X_train, self.y_train = self.datamanager.train_tensors
+
+        if self.datamanager.test_tensors is not None:
+            self.X_test, self.y_test = self.datamanager.test_tensors
+        else:
+            self.X_test, self.y_test = None, None
+
+        self.metric = metric
+        self.task_type = STRING_TO_TASK_TYPES[self.datamanager.task_type]
+        self.output_type = STRING_TO_OUTPUT_TYPES[self.datamanager.output_type]
+        self.issparse = self.datamanager.issparse
+
+        self.seed = seed
+
+        self.output_y_hat_optimization = output_y_hat_optimization
+        # TODO: Check if we need all supported metrics, as in our case even single metric is in a score_dict form
+
+        if isinstance(disable_file_output, (bool, list)):
+            self.disable_file_output = disable_file_output
+        else:
+            raise ValueError('disable_file_output should be either a bool or a list')
+
+        self.model_class: Optional[BaseEstimator] = None
```

Comment:
Maybe `pipeline_class` instead of `model_class` and `DummyClassificationPipeline`/`DummyRegressionPipeline` is more descriptive